### PR TITLE
Fixes newlines in vcvars output for vcvars_dict

### LINKED ISSUE #2904: ValueError exception in client/tools/win.py **bug**
Upgrading to conan 1.3.3 broke my build slave with the following stack trace:

Traceback (most recent call last):
  File "build.py", line 30, in <module>
    builder.run()
  File "E:\Program Files\Python36\lib\site-packages\cpt\packager.py", line 328, in run
    self.run_builds(base_profile_name=base_profile_name)
  File "E:\Program Files\Python36\lib\site-packages\cpt\packager.py", line 404, in run_builds
    r.run()
  File "E:\Program Files\Python36\lib\site-packages\cpt\runner.py", line 48, in run
    with context:
  File "E:\Program Files\Python36\lib\contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "E:\Program Files\Python36\lib\site-packages\conans\client\tools\win.py", line 326, in vcvars
    new_env = vcvars_dict(*args, **kwargs)
  File "E:\Program Files\Python36\lib\site-packages\conans\client\tools\win.py", line 307, in vcvars_dict
    name_var, value = line.split("=", 1)
ValueError: not enough values to unpack (expected 2, got 1)

Details: Conan version 1.3.3, Windows 7, CMake 3.8.2, Visual Studio 2013 Win64

By looking in to it, what happens is line.split("=", 1) on an empty line, in vcvars_dict(). No idea where the empty line comes from (likely from one of VS env scripts), but the following patch solves the symptom. 

```
 conans/client/tools/win.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/conans/client/tools/win.py b/conans/client/tools/win.py
index d025ec4..56dcbbd 100644
--- a/conans/client/tools/win.py
+++ b/conans/client/tools/win.py
@@ -304,8 +304,11 @@ def vcvars_dict(settings, arch=None, compiler_version=None, force=False, filter_
             if "__BEGINS__" in line:
                 start_reached = True
             continue
-        name_var, value = line.split("=", 1)
-        new_env[name_var] = value
+        try:
+            name_var, value = line.split("=", 1)
+            new_env[name_var] = value
+        except (ValueError):
+            pass
 
     if filter_known_paths:
         def relevant_path(path):



```

### PR DESCRIPTION
**Pull request by danimtb**
- [x] Refer to the issue that supports this Pull Request: closes #2904
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [x] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. Also adding a description of the changes in the ``changelog.rst`` file. https://github.com/conan-io/docs

### CODE REVIEW

File: conans/test/util/tools_test.py [hunk 1]
```diff
@@ -625,6 +626,40 @@ def build(self):
             client.run("create . conan/testing")
             self.assertIn("VCINSTALLDIR set to: None", client.out)
 
+    def vcvars_dict_test(self):
+        # https://github.com/conan-io/conan/issues/2904
+        output_with_newline_and_spaces = """__BEGINS__
```
Comment (memsharded) [inline conans/test/util/tools_test.py]: You might need to do ``u"your string content"`` here, for the test to be OK

**Comment by danimtb** [comment]
Already tested it and not working in my python 2.6 setup. Could try in CI if you are convinced...

**Comment by memsharded** [comment]
What is the error?

**Comment by danimtb** [comment]
Same as in Jenkins: `AttributeError: 'str' object has no attribute 'decode'`

File: conans/client/tools/win.py [hunk 2]
```diff
@@ -297,15 +297,22 @@ def vcvars_dict(settings, arch=None, compiler_version=None, force=False, filter_
     cmd = vcvars_command(settings, arch=arch,
                          compiler_version=compiler_version, force=force) + " && echo __BEGINS__ && set"
     ret = decode_text(subprocess.check_output(cmd, shell=True))
+    ret = ret.replace("\n\n", "\n")
```
Comment (lasote) [inline conans/client/tools/win.py]: Really needed? You are checking already if "line"

File: conans/client/tools/win.py [hunk 3]
```diff
@@ -300,12 +300,18 @@ def vcvars_dict(settings, arch=None, compiler_version=None, force=False, filter_
...
                 start_reached = True
             continue
-        name_var, value = line.split("=", 1)
-        new_env[name_var] = value
+        if line == "\n" or not line:
+            continue
+        try:
+            name_var, value = line.split("=", 1)
+            new_env[name_var] = value
+        except ValueError:
```
Comment (memsharded) [inline conans/client/tools/win.py]: Please add test ``value=`` empty, to check
